### PR TITLE
switch `winapi` to `windows` in `_winapi` module

### PR DESCRIPTION
**Pull request by Yaminyam**
#4078

### CODE REVIEW

File: stdlib/src/multiprocessing.rs [hunk 1]
```diff
@@ -4,7 +4,7 @@ pub(crate) use _multiprocessing::make_module;
 #[pymodule]
 mod _multiprocessing {
     use crate::vm::{function::ArgBytesLike, stdlib::os, PyResult, VirtualMachine};
-    use winapi::um::winsock2::{self, SOCKET};
+    use windows::um::winsock2::{self, SOCKET};
```
Comment (youknowone) [inline stdlib/src/multiprocessing.rs:7]: Their api is not the same. manual conversion is required.

**Comment by fanninpm** [comment]
It may be convenient to temporarily modify `.github/workflows/ci.yaml` to focus on the CI operations that run on Windows.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -104,7 +104,7 @@ jobs:
     runs-on:  ${{ matrix.os }}
     strategy:
       matrix:
-        os: [macos-latest, ubuntu-latest, windows-latest]
+        os: [windows-latest]
```
Comment (fanninpm) [inline .github/workflows/ci.yaml:107]:
```suggestion
        os: [macos-latest, ubuntu-latest, windows-latest]
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -206,7 +206,7 @@ jobs:
...
```
Comment (fanninpm) [inline .github/workflows/ci.yaml:209]:
```suggestion
        os: [macos-latest, ubuntu-latest, windows-latest]
```

**Review by fanninpm — COMMENTED** [review]
These suggestions revert the changes done to `ci.yaml` in e119cf5. These should be applied ONLY WHEN THIS IS READY TO MERGE.

Maintainers: this seems like a good candidate for a squash-and-merge.

File: vm/src/stdlib/winapi.rs [hunk 4]
```diff
@@ -4,80 +4,126 @@ pub(crate) use _winapi::make_module;
...
         }
     }
     impl Convertible for i32 {
         fn is_err(&self) -> bool {
             *self == 0
         }
     }
+    impl Convertible for BOOL {
+        fn is_err(&self) -> bool {
+            *self.not().as_bool()
```
Comment (Snowapril) [inline vm/src/stdlib/winapi.rs:106]:
```suggestion
            self.as_bool() == false
```
As we need to import `std::ops::Not` for `not`, its seems better to not use it.

File: vm/src/stdlib/winapi.rs [hunk 5]
```diff
@@ -89,12 +135,15 @@ mod _winapi {
...
+        cvt(vm, unsafe {
+            Foundation::CloseHandle(Foundation::HANDLE(handle as isize))
+        })
+        .map(drop)
     }
 
     #[pyfunction]
     fn GetStdHandle(std_handle: u32, vm: &VirtualMachine) -> PyResult<usize> {
-        cvt(vm, unsafe { processenv::GetStdHandle(std_handle) }).map(husize)
+        cvt(vm, Console::GetStdHandle(STD_HANDLE(std_handle))).map(husize)
```
Comment (Snowapril) [inline vm/src/stdlib/winapi.rs:146]:
```suggestion
        let handle = unsafe { Console::GetStdHandle(STD_HANDLE(std_handle)) }.map_err(|err| vm.new_type_error(format!("{}", err.message()) ))?;
        cvt(vm, handle).map(husize)
```
This is one kind of example. please convert error type properly.

File: vm/src/stdlib/winapi.rs [hunk 6]
```diff
@@ -177,16 +227,18 @@ mod _winapi {
         args: CreateProcessArgs,
         vm: &VirtualMachine,
     ) -> PyResult<(usize, usize, u32, u32)> {
-        let mut si = winbase::STARTUPINFOEXW::default();
+        let mut si = Threading::STARTUPINFOEXW::default();
         si.StartupInfo.cb = std::mem::size_of_val(&si) as _;
 
         macro_rules! si_attr {
```
Comment (Snowapril) [inline vm/src/stdlib/winapi.rs:234]: For using tuple struct, we need to modify original macro like below. Please improve it as you want
```rust
macro_rules! si_attr {
            ($attr:ident, $t:ty) => {{
                si.StartupInfo.$attr = <Option<$t>>::try_from_object(
                    vm,
                    args.startup_info.get_attr(stringify!($attr), vm)?,
                )?
                .unwrap_or(0)
            }};
            ($attr:ident, $t:ty, $tt:path) => {{
                si.StartupInfo.$attr = $tt(<Option<$t>>::try_from_object(
                    vm,
                    args.startup_info.get_attr(stringify!($attr), vm)?,
                )?
                .unwrap_or(0));
            }};
        }
        si_attr!(dwFlags, u32, STARTUPINFOW_FLAGS);
        si_attr!(wShowWindow, u16);
        si_attr!(hStdInput, isize, HANDLE);
        si_attr!(hStdOutput, isize, HANDLE);
        si_attr!(hStdError, isize, HANDLE);
```

File: vm/src/stdlib/winapi.rs [hunk 7]
```diff
@@ -235,29 +289,29 @@ mod _winapi {
...
+                windows::core::PWSTR(command_line),
                 null_mut(),
                 null_mut(),
-                args.inherit_handles,
+                BOOL(args.inherit_handles),
                 args.creation_flags
-                    | winbase::EXTENDED_STARTUPINFO_PRESENT
-                    | winbase::CREATE_UNICODE_ENVIRONMENT,
+                    | Threading::EXTENDED_STARTUPINFO_PRESENT
+                    | Threading::CREATE_UNICODE_ENVIRONMENT,
```
Comment (Snowapril) [inline vm/src/stdlib/winapi.rs:301]: As we cant or operation between `u32` and `struct(u32)`, we need to do like this
```rust
PROCESS_CREATION_FLAGS(args.creation_flags
                    | Threading::EXTENDED_STARTUPINFO_PRESENT.0
                    | Threading::CREATE_UNICODE_ENVIRONMENT.0)
```

**Pull request closed without merging**